Favor/Favor/Sources/Scenes/Profile/FriendPage/Reactors/FriendPageViewReactor.swift

```diff
@@ -0,0 +1,216 @@
+      ])
+      
+    case .moreAnniversaryDidTap:
+      self.steps.accept(AppStep.anniversaryListIsRequired)
+      return .empty()
+      
+    case .doNothing:
+      return .empty()
+    }
+  }
+  
+  func reduce(state: State, mutation: Mutation) -> State {
+    var newState = state
+    
+    switch mutation {
+    case .setMemoSection(let memo):
+      newState.memoItems = [.memo(memo)]
+      
+    case .setFavorSection(let favors):
+      let favorSection = favors.map {
+        ProfileSectionItem.favors(ProfileFavorCellReactor(favor: $0))
+      }
+      newState.favorItems = favorSection
+      
+    case .setAnniversarySection(let anniversaries):
+      let anniversarySection = anniversaries.map {
+        ProfileSectionItem.anniversaries(ProfileAnniversaryCellReactor(anniversary: $0))
+      }
+      newState.anniversaryItems = anniversarySection
+      
+    case .setMemo(let memo):
+      newState.friendMemo = memo
+      
+    case .setFriendName(let name):
+      newState.friendName = name
+      
+    case .setLoading(let isLoading):
+      newState.isLoading = isLoading
+    }
+    
+    return newState
+  }
+  
+  func transform(state: Observable<State>) -> Observable<State> {
+    return state.map { state in
+      var newState = state
+      var newSection: [ProfileSection] = []
+      var newItems: [[ProfileSectionItem]] = []
+      
+      // 새 기념일 도움 섹션
+      // 유저가 아닌 친구 & 기념일 목록이 비어 있는 조건을 모두 만족해야합니다.
+      if !self.isUser, self.friend.anniversaryList.toArray().isEmpty {
+        newSection.append(.anniversarySetupHelper)
+        newItems.append([.anniversarySetupHelper])
+      }
+      
+      // 취향
+      if !state.favorItems.isEmpty {
+        newSection.append(.favors)
+        newItems.append(state.favorItems)
+      }
+      
+      // 기념일
+      if !state.anniversaryItems.isEmpty {
+        newSection.append(.anniversaries)
+        newItems.append(state.anniversaryItems)
+      }
+      
+      // 메모
+      newSection.append(.memo)
+      newItems.append(state.memoItems)
+      
+      newState.sections = newSection
+      newState.items = newItems
+      return newState
+    }
+  }
+}
+
+private extension FriendPageViewReactor {
+  func setupFriendFetcher() {
+    // onRemote
+    self.friendFetcher.onRemote = {
+      let networking = FriendNetworking()
+      let friend = networking.request(.patchFriend(
+        friendName: self.currentState.friendName,
+        friendMemo: self.currentState.friendMemo ?? "",
+        friendNo: self.friend.friendNo
+      ))
+        .flatMap { response -> Observable<[Friend]> in
+          let friend: ResponseDTO<FriendResponseDTO> = try APIManager.decode(response.data)
+          return .just([friend.data.toDomain()])
+        }
+        .asSingle()
+      return friend
+    }
+    // onLocal
+    self.friendFetcher.onLocal = {
+      return try await RealmManager.shared.read(Friend.self).filter(
+          "friendNo == %@",
+          self.friend.friendNo
+      )
```

Comment:
Realm 관련된 쿼리를 할 때는 Realm에서 제공하는 `where` 문을 사용하는 게 좋을 듯..!
```suggestion
      return try await RealmManager.shared.read(Friend.self).where {
        $0.friendNo == self.friend.friendNo
      }
```

Reply:
혹시 이 코드 에러 뜨는데.. 클로저의 `return` 타입을 `Query<Bool>`로 하라고 하는데, 
`Query<T>`는 생성자가 `internal`로 되어있어서 어떻게 해야할지 감이 안 잡히네..
혹시 방법 아는 게 있을까?

Reply:
흠.. 나는 `friendNo`가 `Int` 타입이 아니라는 이상한 에러가 뜨네..
안그래도 Realm 관련해서 손 볼곳들이 좀 있어서 작업해보고 해결방법 있으면 알려줄게..!

``` Swift
return try await RealmManager.shared.read(Friend.self).where {
  $0.friendNo.in([self.friend.friendNo])
}
```

이렇게 사용하면 아마 되긴 할거야..!